### ISSUE TITLE
Cross-compilation of SDL2 failed for QNX

### ISSUE THREAD

**Issue by asad26** [libsdl-org/SDL#8321]
I am cross compiling SDL2 for Qualcomm QNX on Ubuntu Focal, which is failing with the following error at CMake generate step:

```
CMake Error at cmake/macros.cmake:36 (message):
  *** ERROR: Threads are needed by many SDL subsystems and may not be
  disabled
Call Stack (most recent call first):
  CMakeLists.txt:2965 (message_error)
```

I used SDL2 branch (commit hash: `6b5eadb10f02b51b400e710e2c95e9f0c2cb579c`) of SDL repo and this is the CMake toolchain file `aarch64-qnx.cmake`:

```
set(CMAKE_SYSROOT $ENV{QNX_TARGET})
set(CMAKE_SYSTEM_NAME QNX)
set(CMAKE_SYSTEM_VERSION 710)
set(CMAKE_SYSTEM_PROCESSOR "aarch64")
set(arch nto${CMAKE_SYSTEM_PROCESSOR})

set(CMAKE_AS "$ENV{QNX_HOST}/usr/bin/${arch}-as")
set(CMAKE_AR "$ENV{QNX_HOST}/usr/bin/${arch}-ar")
set(CMAKE_C_COMPILER "$ENV{QNX_HOST}/usr/bin/${arch}-gcc")
set(CMAKE_CXX_COMPILER "$ENV{QNX_HOST}/usr/bin/${arch}-g++")
set(CMAKE_LINKER "$ENV{QNX_HOST}/usr/bin/${arch}-ld")

set(CMAKE_C_COMPILER_TARGET "gcc_ntoaarch64le")
set(CMAKE_CXX_COMPILER_TARGET "gcc_ntoaarch64le")
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fexceptions -fPIC -D_XOPEN_SOURCE=600")
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fexceptions -fPIC -D_XOPEN_SOURCE=600 -std=gnu++14")

add_definitions(-D_POSIX_C_SOURCE=200112L -D_QNX_SOURCE)

set(CMAKE_FIND_LIBRARY_PREFIXES "lib")
set(CMAKE_FIND_LIBRARY_SUFFIXES ".so;.a")

set(CMAKE_FIND_ROOT_PATH "${CMAKE_SYSROOT}")
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
```

The CMake version is `3.26.1`. 

CMake generate command:

`cmake -B build -DCMAKE_TOOLCHAIN_FILE=$(pwd)/aarch64-qnx.cmake -DSDL_X11=0 -DCMAKE_BUILD_TYPE=Release`

Any help would be much appreciated. Thanks.

**Comment by madebr** [comment]
SDL2's cmake does not currently have any qnx support.
https://github.com/libsdl-org/SDL/pull/8322 is addressing this.
Can you try that one, and also provide fixes? (I'm no qnx user)

**Comment by madebr** [comment]
This has been fixed in #8322

**Comment by asad26** [comment]
Ok thanks. 
Let me try again.

**Comment by asad26** [comment]
@madebr 
I can confirm that the latest version of SDL2 branch builds by setting the following environment variables:

```
set(ENV{PKG_CONFIG_LIBDIR} "/do_not_use_pkg_config_when_cross_compiling")
set(ENV{PKG_CONFIG_PATH} "/do_not_use_pkg_config_when_cross_compiling")
```
